src/index.js

```diff
@@ -68,6 +88,46 @@ const init = async () => {
         },
     });
 
+    server.route({
+        method: 'POST',
+        path: '/donation',
+        handler: require('./handlers/donation/post'),
+        options: {
+            validate: {
+                payload: Joi.object({
+                    method: Joi.string().valid('mollie', 'creditcard', 'cryptocurrency').required(),
+                    amount: Joi.string()
+                        .pattern(/^[0-9]+\.[0-9]{2}/)
+                        .required(),
+                    description: Joi.string(),
+                    reference: Joi.string().required(),
+                    redirect_base: Joi.string()
+                        .required()
+                        .pattern(/^https:\/\/.*/),
+                }),
+            },
+        },
+    });
+
+    server.route({
+        method: 'GET',
+        path: '/donation/state/{reference*}',
+        handler: require('./handlers/donation/state'),
+        options: {
+            validate: {
+                params: Joi.object({
+                    reference: Joi.string().required(),
+                }),
+            },
+        },
+    });
+
+    server.route({
+        method: 'GET',
```

Comment:
This should really be POST. GET is not supposed to change the state.

```suggestion
        method: 'POST',
```